inc/plugins/myalerts.php

```diff
@@ -859,26 +859,36 @@ function myalerts_datahandler_user_insert(&$dataHandler)
 	);
 }
 
+$plugins->add_hook(
+	'fetch_wol_activity_end',
+	'myalerts_online_activity'
+);
+function myalerts_online_activity($user_activity)
+{
+	if ($user_activity['activity'] == 'unknown') {
+		$split_loc = explode('.php', $user_activity['location']);
+		$filename = my_substr($split_loc[0], -my_strpos(strrev($split_loc[0]), '/'));
```

Comment:
This may be easier to do with `basename` if I'm understanding the logic correctly?

Reply:
Yep, I shamelessly ripped those lines from lines [#26](https://github.com/mybb/mybb/blob/a4dc072b3f4a9beed0d0686ffa6b6c0d1e343d59/inc/functions_online.php#L26) and [#33](https://github.com/mybb/mybb/blob/a4dc072b3f4a9beed0d0686ffa6b6c0d1e343d59/inc/functions_online.php#L33) of `inc/functions_online.php` after confirming that the logic was correct. We *could* simplify the code a little with your suggestions, although there'd be some admittedly only hypothetical failure cases for locations of, e.g., "alerts.php/test?abc=def", and "rootpluginscript.php/alerts". They're only hypothetical 'cos they don't seem like plausible returns of `get_current_location()` in `inc/functions.php`, so I guess it's a choice between purity (avoiding unrealisable failure cases) and simplicity...

Reply:
OK, I hadn't realised it as copied out of the core. In that case, we'll leave it as it is.